src/runtime/virtcontainers/qemu_arch_base.go

```diff
@@ -43,6 +43,15 @@ type qemuArch interface {
 	// machine returns the machine type
 	machine() govmmQemu.Machine
 
+	// protection returns guest protection type
+	guestProtection() guestProtection
+
+	// set attestation id for SEV guests with pre-attestation
+	setSEVAttestationId(attestation_id string)
+
+	// set attestation id for SEV guests with pre-attestation
+	getSEVAttestationId() string
```

Comment:
1. comment describes `set` instead of `get`
2. function name -- other functions don't have a `get` prefix, so maybe this should be `sevAttestationId()`